src/Reading/LengthAttributeConverter.php

```diff
@@ -30,10 +30,7 @@ private function __construct()
      */
     public static function getInstance(): self
     {
-        if (!isset(self::$instance)) {
-            self::$instance = new self();
-        }
-        return self::$instance;
+        return self::$instance ??= new self();
```

Comment:
Can we split this in two? While compact, I think assignments should always be their own statements if possible.

```suggestion
        self::$instance ??= new self();
        return self::$instance;
```